crypto/fipsmodule/dh/check.c

```diff
@@ -203,24 +202,6 @@ int DH_check(const DH *dh, int *out_flags) {
     if (!BN_is_one(t2)) {
       *out_flags |= DH_CHECK_INVALID_Q_VALUE;
     }
-  } else if (BN_is_word(dh->g, DH_GENERATOR_2)) {
-    l = BN_mod_word(dh->p, 24);
-    if (l == (BN_ULONG)-1) {
-      goto err;
-    }
-    if (l != 11) {
-      *out_flags |= DH_CHECK_NOT_SUITABLE_GENERATOR;
-    }
-  } else if (BN_is_word(dh->g, DH_GENERATOR_5)) {
-    l = BN_mod_word(dh->p, 10);
-    if (l == (BN_ULONG)-1) {
-      goto err;
-    }
-    if (l != 3 && l != 7) {
```

Comment:
what's significant about these values or 11 (mod 24) for `g=2`? why did we previously restrict `p` to these values for relevant `g`?

Reply:
Details [here](https://wiki.openssl.org/index.php/Diffie-Hellman_parameters#Validating_Parameters).